float32[] fdbk_type
float32 input_type